dbms/src/Storages/StorageMySQL.cpp

```diff
@@ -56,6 +64,99 @@ BlockInputStreams StorageMySQL::read(
 }
 
 
+class StorageMySQLBlockOutputStream : public IBlockOutputStream
+{
+public:
+    explicit StorageMySQLBlockOutputStream(
+            const StorageMySQL & storage,
+            const std::string & remote_database_name,
+            const std::string & remote_table_name ,
+            const mysqlxx::PoolWithFailover::Entry & entry,
+            const size_t & mysql_max_rows_to_insert)
+        : sample_block{storage.getSampleBlock()}
+        , remote_database_name{remote_database_name}
+        , remote_table_name{remote_table_name}
+        , entry{entry}
+        , max_batch_rows{mysql_max_rows_to_insert}
+    {
+    }
+
+   Block getHeader() const override { return sample_block; }
+
+   void write(const Block & block) override
+   {
+       auto blocks = splitBlocks(block, max_batch_rows);
+       mysqlxx::Transaction trans(entry);
```

Comment:
Maybe inserting in single transaction should be optional.
Sometimes user is going to export huge amount of data and to see while it loads.